src/lib/comms/sol-oic-cbor.c

```diff
@@ -63,7 +63,7 @@ initialize_cbor_payload(struct sol_oic_map_writer *encoder)
 
     cbor_encoder_init(&encoder->encoder, encoder->payload,
         buf->capacity - offset, 0);
-    old_ptr = encoder->encoder.ptr;
+    old_ptr = encoder->encoder.data.ptr;
```

Comment:
You're supposed to use cbor_encoder_get_used_bytes and add to the original pointer.